File: services/core/entity/task_input.go
package entity

import (
	"time"

	"tenkhours/pkg/types"
)

type TaskInput struct {
	ID            *string         `json:"id,omitempty"`
	CategoryID    *string         `json:"categoryID,omitempty"`
	Name          string          `json:"name"`
	Priority      int             `json:"priority"`
	Subtasks      []CheckboxInput `json:"subtasks"`
	Description   *string         `json:"description"`
	CompletedTime *time.Time      `json:"completedTime"`
	Deadline      *time.Time      `json:"deadline"`
}

type TaskPipeline struct {
	Filter  *TaskFilter
	OrderBy *TaskOrderBy
	*types.Pagination
}

type TaskFilter struct {
	CharacterID  *string  `json:"character_id"`
	CharacterIDs []string `json:"character_ids"`
	IsCompleted  *bool    `json:"is_completed"`
}

type TaskOrderBy struct {
	Priority *types.SortOrder `json:"priority"`
}

File: services/penk/src/utils/grpc/proto/core/TaskInput.ts
// Original file: ../../proto/core/task_message.proto

import type {
  CheckboxInput as _core_CheckboxInput,
  CheckboxInput__Output as _core_CheckboxInput__Output,
} from "./CheckboxInput";
import type { Long } from "@grpc/proto-loader";

export interface TaskInput {
  id?: string;
  categoryId?: string;
  name?: string;
  priority?: number;
  completedTime?: number | string | Long;
  subtasks?: _core_CheckboxInput[];
  description?: string;
  deadline?: number | string | Long;
  _id?: "id";
  _categoryId?: "categoryId";
  _completedTime?: "completedTime";
  _description?: "description";
  _deadline?: "deadline";
}

export interface TaskInput__Output {
  id?: string;
  categoryId?: string;
  name: string;
  priority: number;
  completedTime?: string;
  subtasks: _core_CheckboxInput__Output[];
  description?: string;
  deadline?: string;
  _id: "id";
  _categoryId: "categoryId";
  _completedTime: "completedTime";
  _description: "description";
  _deadline: "deadline";
}
